Benchmark differenciation (#3133)
geodynamics/aspect: doc/manual/manual.tex

\subsubsection{Compaction -- differentiation}

\textit{This section was contributed by Marine Lasbleis.}

> Here we use \aspect{} to mode the compaction of a simple column with a mixture of liquid and solid using two-phase flow equations. 

Reviewer: ```suggestion
Here we use \aspect{} to model the compaction of a simple column with a mixture of liquid and solid using two-phase flow equations. 
```

It presents a modification of the global melt material model to use different parametrization of the permeability, shear viscosity and bulk viscosity. 

We model the compaction of a mixture of liquid and fluid, without additional melting and freezing.  
This set-up has been used in the PhD thesis of Ondrej Sramek (\cite{SRPHD07}) as an analog for the differentiation of an initially undifferentiated planet where the solid matrix is the silicates and the liquid the molten iron. 
We test here using the same set of equations than in \cite{SRB07} (SRB07), but without melting and freezing and in the limit of $\omega=0$
and the liquid viscosity negligible compared to the solid one. 
Both phases are considered to be viscously deformed by the compaction, and the liquid is lighter than the solid. 

> The plugin and parameter files can be found at \url{benchmarks/differenciation/melt_srb.cc} and \url{benchmarks/differenciation/diff_adim.prm}. 

Reviewer: need to adjust the directory names here and in the line below

Author: done

A jupyter notebook (Python 3) for recreating the different figures shown here is provided in \url{benchmarks/differenciation/python}. 

\paragraph{Modification of the MeltGlobal material model}

To solve the SRB07 equations with the melt model implemented in \aspect{}, 
we need to modify the formulations for the shear and bulk viscosity ($\eta$ and $\xi$) 
and the Darcy coefficient $K_D$ to use the viscosities of the solid $\mu_m$ and the liquid $\mu_f$, as  

\begin{equation}
    K_D = \frac{k(\phi)}{\mu_f} = \frac{k_0 \phi^2}{\mu_f}
\end{equation}
\begin{equation}
    \eta = (1-\phi) \mu_m,
\end{equation}
\begin{equation}
    \xi = \frac{1-\phi}{\phi} K \mu_m
\end{equation}

The compaction length is then calculated as 
\begin{equation}
    \delta = \sqrt{\frac{\mu_m}{\mu_f}k_0} 
\end{equation}
and is constant with the porosity. 

> The global melt model is set up with compressibility, which we will need to remove by setting the parameters to 0. 

Reviewer: ```suggestion
The global melt model is originally set up with compressibility, which we will need to remove by setting the corresponding parameters to 0. 
```

We are also neglecting the effects of temperature on the parameters and setting the temperature field to 0 everywhere. 

> The viscosities tends to 0 or infinity around $\phi=0$ and $\phi=1$, which is not possible to implement in numerical model. 

Reviewer: ```suggestion
The viscosities tends to 0 or infinity around $\phi=0$ and $\phi=1$, which is not possible to implement in numerical models. 
```

> We set a maximum viscosity variation due to porosity of $10^6$.

Reviewer: ```suggestion
We therefore set a maximum viscosity variation due to porosity of $10^6$.
```

We defined a new material model which incorporate all of these modifications, based on the material model global melt. 

> The plugin \url{benchmarks/differenciation/melt_srb.cc} is used as an additional library and needs to be compiled befre running the parameter file. 

Reviewer: ```suggestion
The plugin \url{benchmarks/differenciation/melt_srb.cc} is used as an additional library and needs to be compiled before running the parameter file. 
```

> \paragraph{Set-up}

Reviewer: ```suggestion
\paragraph{Model Setup}
```

We set a column of width one cell, with 200 cells in the y-direction. 

> The total size of the system is 1 and all parameters are put to 0 or 1 to have a dimensionless setting. 

Reviewer: ```suggestion
The total area of the system is 1 and all parameters are put to 0 or 1 to have a dimensionless setting. 
```

Author: it's the total height. Modified.

The gravity is constant. 
The bottom boundary is set at porosity of zero and no slip boundary. 
The top boundary is set at a porosity of 1, and no flows are permitted (the gradient of fluid pressure at the top is set to the fluid weight). 
We use a compaction length of 1 and 0.2 for this benchmark (ones need to set the fluid viscosity to 25 to change this value from 1 to 0.2), 
and we start with a constant porosity profile of 30\%.

> The parameter file is available at \url{benchmarks/differenciation/diff_adim.prm}. 

Reviewer: change directory name here

\paragraph{First time step}

The first time step solves the velocity equation for a constant porosity. 
The 1-D equations with constant porosity $\phi_0$ have an analytical solution for the matrix velocity given by  \cite{SRPHD07} as
\begin{equation}
    v_m(y) = -\delta ^2 \phi_0^2 (1-\phi_0)\left [ 1+\frac{\sinh\left ( \frac{1}{h}\right )\sinh\left (\frac{y}{h}\right )}{\cosh\left ( \frac{1}{h}\right ) +1} - \cosh\left ( \frac{y}{h}\right )\right ], 
\end{equation}
where $h$ is length scale related to the compaction length as 
\begin{equation}
    h=\delta \sqrt{\phi_0 \left (1+\frac{4}{3}\phi_0 \right )(1-\phi_0)}.
\end{equation}

Comparison between the analytical solution and our runs for two compaction lengths (1 and 0.2) are shown on figure \ref{fig:analytic}. 

> The agreement is good, with a slight discrepancy at the top, likely due to the top boundary condition not exactly defines at $y=1$.   

Reviewer: ```suggestion
The agreement is good, with a slight discrepancy at the top, likely due to the top boundary condition not exactly defined at $y=1$.   
```

\begin{figure}[h]
    \centering
    \includegraphics[width=0.49\textwidth]{cookbooks/benchmarks/delta_1_analytical.pdf}
    \hfill
    \includegraphics[width=0.49\textwidth]{cookbooks/benchmarks/delta_02_analytical.pdf}
    \caption{Comparison between analytical solution (yellow x symbols) and velocity of the matrix at the first time step (blue solid line), for a compaction length of 1 (left figure) and 0.2 (right figure).}
    \label{fig:analytic}
\end{figure}

\paragraph{Evolution}

> The porosity profiles evolves such that the liquid is expelled to the top and the matrix compacts at the bottom. 

Reviewer: ```suggestion
The porosity profile evolves such that the liquid is expelled to the top and the matrix compacts at the bottom. 
```

No flow can cross the top or bottom boundaries. 
We first check the mass conservation of the liquid with time, for different number of points in the domain. 
Results are shown on figure \ref{fig:comp}. 
There is an overshoot at the beginning, likely due to the formation of the wiggles, and then depending on the number of points used, the system tends to stabilize.  
Results seem to show that at least 200 points are required for this case of $\delta = 1$. 

\begin{figure}
    \centering
    \includegraphics[width=0.5\textwidth]{cookbooks/benchmarks/comparison_pts.pdf}
    \caption{Evolution of the total mass of liquid (normalized by the initial mass) for different discretization, with a compaction length $\delta=1$. }
    \label{fig:comp}
\end{figure}

The porosity and matrix velocity evolution are shown on figure \ref{fig:evolution}. 

> Results are similar to \cite{SRPHD07}, except for porosity close to 1, where \aspect{} do not converge correctly and overestimate the segregation. 

Reviewer: ```suggestion
Results are similar to \cite{SRPHD07}, except for porosity close to 1, where \aspect{} does not converge correctly and overestimates the segregation. 
```

> Since the model was designed to solve for very low porosity, and that the equations should not be physically valid for porosity high enough than the matrix is no longer connected (the highest threshold usually considered is about 40\%, corresponding to a system made of spheres with a random close packing)

Reviewer: This sentence is not complete. You likely want to say the above deviation is expected, or at least explainable?

Author: modified

\subsubsection{Compaction in a column with melt}

\textit{This section was contributed by Marine Lasbleis.}

Here we use \aspect{} to model the compaction of a simple column with a mixture of liquid and solid using two-phase flow equations. 
It presents a modification of the global melt material model to use different parametrization of the permeability, shear viscosity and bulk viscosity. 

We model the compaction of a mixture of liquid and fluid, without additional melting and freezing.  
This set-up has been used in the PhD thesis of Ondrej Sramek (\cite{SRPHD07}) as an analog for the differentiation of an initially undifferentiated planet where the solid matrix is the silicates and the liquid the molten iron. 
We test here using the same set of equations than in \cite{SRB07} (SRB07), but without melting and freezing and in the limit of $\omega=0$
and the liquid viscosity negligible compared to the solid one. 
Both phases are considered to be viscously deformed by the compaction, and the liquid is lighter than the solid. 

> The plugin and parameter files can be found at \url{benchmarks/melt_column_differenciation/melt_SramekRicardBercovici2007.cc} and \url{benchmarks/melt_column_differenciation/diff_adim.prm}. 

Reviewer: need to adjust the directory name

> A jupyter notebook (Python 3) for recreating the different figures shown here is provided in \url{benchmarks/melt_column_differenciation/python}. 

Reviewer: and here

we need to modify the formulations for the shear and bulk viscosity ($\eta$ and $\xi$) 
and the Darcy coefficient $K_D$ to use the viscosities of the solid $\mu_m$ and the liquid $\mu_f$, as  

\begin{equation}
    K_D = \frac{k(\phi)}{\mu_f} = \frac{k_0 \phi^2}{\mu_f}
\end{equation}
\begin{equation}
    \eta = (1-\phi) \mu_m,
\end{equation}
\begin{equation}
    \xi = \frac{1-\phi}{\phi} K \mu_m
\end{equation}

The compaction length is then calculated as 
\begin{equation}
    \delta = \sqrt{\frac{\mu_m}{\mu_f}k_0} 
\end{equation}
and is constant with the porosity. 

The global melt model is originally set up with compressibility, which we will need to remove by setting the corresponding parameters to 0. 
We are also neglecting the effects of temperature on the parameters and setting the temperature field to 0 everywhere. 
The viscosities tends to 0 or infinity around $\phi=0$ and $\phi=1$, which is not possible to implement in numerical models. 
We therefore set a maximum viscosity variation due to porosity of $10^6$.

We defined a new material model which incorporate all of these modifications, based on the material model global melt. 

> The plugin \url{benchmarks/melt_column_differenciation/melt_srb.cc} is used as an additional library and needs to be compiled before running the parameter file. 

Reviewer: and here

\paragraph{Model Setup}

We set a column of width one cell, with 200 cells in the y-direction. 
The total height of the system is 1 and all parameters are put to 0 or 1 to have a dimensionless setting. 
The gravity is constant. 
The bottom boundary is set at porosity of zero and no slip boundary. 
The top boundary is set at a porosity of 1, and no flows are permitted (the gradient of fluid pressure at the top is set to the fluid weight). 
We use a compaction length of 1 and 0.2 for this benchmark (ones need to set the fluid viscosity to 25 to change this value from 1 to 0.2), 
and we start with a constant porosity profile of 30\%.

> The parameter file is available at \url{benchmarks/melt_column_differenciation/diff_adim.prm}. 

Reviewer: and here

    \label{fig:analytic}
\end{figure}

\paragraph{Evolution}

The porosity profile evolves such that the liquid is expelled to the top and the matrix compacts at the bottom. 
No flow can cross the top or bottom boundaries. 
We first check the mass conservation of the liquid with time, for different number of points in the domain. 
Results are shown on figure \ref{fig:comp}. 
There is an overshoot at the beginning, likely due to the formation of the wiggles, and then depending on the number of points used, the system tends to stabilize.  
Results seem to show that at least 200 points are required for this case of $\delta = 1$. 

\begin{figure}
    \centering
    \includegraphics[width=0.5\textwidth]{cookbooks/benchmarks/comparison_pts.pdf}
    \caption{Evolution of the total mass of liquid (normalized by the initial mass) for different discretization, with a compaction length $\delta=1$. }
    \label{fig:comp}
\end{figure}

The porosity and matrix velocity evolution are shown on figure \ref{fig:evolution}. 
Results are similar to \cite{SRPHD07}, except for porosity close to 1, where \aspect{} does not converge correctly and overestimates the segregation. 

> The model was designed to solve for very low porosity, and the equations are not be physically valid for porosity high enough -- the matrix is then no longer connected -- then this result was expected.  

Reviewer: ```suggestion
The model was designed to solve for very low porosity, and the equations are not be physically valid for porosity high enough -- the matrix is then no longer connected -- therefore this result was expected.  
```
